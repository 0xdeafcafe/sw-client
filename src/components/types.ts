export type DisplayProps = {
  currentRoot: string;
  item: FilmItemProps &
    PersonItemProps &
    PlanetItemProps &
    SpeciesItemProps &
    StarshipItemProps &
    VehicleItemProps;
};

export type FilmProps = {
  item: FilmItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type FilmItemProps = {
  characters: string[];
  created: string;
  director: string;
  edited: string;
  episode_id: number;
  opening_crawl: string;
  planets: string[];
  producer: string;
  release_date: string;
  species: string[];
  starships: string[];
  title: string;
  url: string;
  vehicles: string[];
};

export type PersonProps = {
  item: PersonItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type PersonItemProps = {
  created: string;
  edited: string;
  birth_year: string;
  eye_color: string;
  films: string[];
  gender: string;
  hair_color: string;
  height: string;
  homeworld: string;
  name: string;
  skin_color: string;
  species: string[];
  starships: string[];
  url: string;
  vehicles: string[];
  mass: string;
};

export type PlanetProps = {
  item: PlanetItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type PlanetItemProps = {
  name: string;
  diameter: string;
  orbital_period: string;
  population: string;
  rotation_period: string;
  surface_water: string;
  gravity: string;
  terrain: string;
  climate: string;
  films: string[];
  residents: string[];
  url: string;
};

export type SpeciesProps = {
  item: SpeciesItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type SpeciesItemProps = {
  name: string;
  average_height: string;
  average_lifespan: string;
  classification: string;
  designation: string;
  eye_colors: string;
  hair_colors: string;
  language: string;
  skin_colors: string;
  films: string[];
  people: string[];
  url: string;
};

export type StarhsipProps = {
  item: StarshipItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type StarshipItemProps = {
  name: string;
  model: string;
  manufacturer: string;
  cargo_capacity: string;
  hyperdrive_rating: string;
  consumables: string;
  cost_in_credits: string;
  crew: string;
  length: string;
  max_atmosphering_speed: string;
  passengers: string;
  starship_class: string;
  films: string[];
  pilots: string[];
  url: string;
};

export type VehicleProps = {
  item: VehicleItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type VehicleItemProps = {
  name: string;
  model: string;
  manufacturer: string;
  cargo_capacity: string;
  consumables: string;
  cost_in_credits: string;
  crew: string;
  length: string;
  max_atmosphering_speed: string;
  passengers: string;
  vehicle_class: string;
  films: string[];
  pilots: string[];
  url: string;
};

export type AccordionProps = {
  item: FilmItemProps &
    PersonItemProps &
    PlanetItemProps &
    SpeciesItemProps &
    StarshipItemProps &
    VehicleItemProps;
  currentRoot: string;
};

export type ModalProps = {
  displayName: String;
  result: () => React.ReactNode;
  item: string[];
};

export type State = {
  roots: {
    isLoading: boolean;
    isBranchedRootLoading: boolean;
    payload: object;
    error: string[];
    currentRoot: StateCurrentRoot;
    branchedRoot: FilmItemProps[] &
      PersonItemProps[] &
      PlanetItemProps[] &
      SpeciesItemProps[] &
      StarshipItemProps[] &
      VehicleItemProps[];
  };
};

type StateCurrentRoot = {
  count: number;
  next: string;
  previous: string | null;
  results: FilmItemProps[] &
    PersonItemProps[] &
    PlanetItemProps[] &
    SpeciesItemProps[] &
    StarshipItemProps[] &
    VehicleItemProps[];
  k: string;
};
