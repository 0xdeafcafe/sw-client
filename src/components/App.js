import React from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

const App = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
