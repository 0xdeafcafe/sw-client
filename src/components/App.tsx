
import React, {ReactNode} from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

interface AppProps {
	children: ReactNode;
  }

const App: React.FC<AppProps> = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
