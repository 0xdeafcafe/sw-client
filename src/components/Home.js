import classnames from 'classnames';
import { fetchRoots } from '../actions';
import sentenceCase from 'sentence-case';
import {
	Nav,
	NavItem,
	NavLink,
	TabContent,
	TabPane,
} from 'reactstrap';
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

const Home = () => {
	const dispatch = useDispatch();
	const [tab, setTab] = useState(void 0);
	const roots = useSelector(state => state.roots);

	useEffect(() => {
		dispatch(fetchRoots());
	}, [dispatch]);

	const keys = Object.keys(roots.payload || {});

	return (
		<div>
			<h1>{'My little Star Wars app ðŸ‘¾'}</h1>

			{roots.payload && (
				<div className={'mt-3'}>
					<Nav tabs>
						{keys.map(k => (
							<NavItem key={k}>
								<NavLink
									className={classnames({ active: tab === k })}
									onClick={() => setTab(k)}
								>
									{sentenceCase(k)}
								</NavLink>
							</NavItem>
						))}
					</Nav>

					<TabContent activeTab={tab}>
						{keys.map(k => (
							<TabPane
								key={k}
								tabId={k}
							>
								<code>{'todo: load resource table'}</code>
							</TabPane>
						))}
					</TabContent>
				</div>
			)}
		</div>
	);
};

export default Home;
