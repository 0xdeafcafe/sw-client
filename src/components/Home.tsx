import classnames from "classnames";
import sentenceCase from "sentence-case";
import styled from "styled-components";
import { fetchIndividualRoot, fetchRoots } from "../actions/actions";
import { Nav, NavItem, NavLink, TabContent, TabPane, Button } from "reactstrap";
import CircularProgress from "@mui/material/CircularProgress";

import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { ItemAccordion } from "./Accordion/Accordion";
import { FilmItemProps, PersonItemProps, PlanetItemProps, SpeciesItemProps, StarshipItemProps, VehicleItemProps } from "./types";

const SpinnerContainer = styled.div`
  display: flex;
  justify-content: center;
  padding: 50px;
`;

const StyledButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`

const Home = () => {
  const dispatch = useDispatch();
  const [tab, setTab] = useState<string | undefined>(void 0);

  const roots = useSelector((state: any) => state.roots);
  const isLoading = useSelector((state: any) => state.roots.isLoading);
  const currentRoot = useSelector(
    (state: any) =>
      (state &&
        state.roots &&
        state.roots.currentRoot &&
        state.roots.currentRoot.k) ||
      []
  );

  const currentRootResults = useSelector(
    (state: any) =>
      (state &&
        state.roots &&
        state.roots.currentRoot &&
        state.roots.currentRoot) ||
      []
  );

  useEffect(() => {
    dispatch(fetchRoots());
  }, [dispatch]);

  const keys = Object.keys(roots.payload || {});

  const handleTabClick = (k: string) => {
    dispatch(fetchIndividualRoot(k));
  };
  return (
    <div>
      <h1>{"My little Star Wars app ðŸ‘¾"}</h1>

      {roots.payload && (
        <div className={"mt-3"}>
          <Nav tabs>
            {keys.map((k) => (
              <NavItem key={k}>
                <NavLink
                  className={classnames({ active: tab === k })}
                  onClick={() => {
                    setTab(k);
                    handleTabClick(k);
                  }}
                >
                  {sentenceCase(k)}
                </NavLink>
              </NavItem>
            ))}
          </Nav>

          {isLoading ? (
            <SpinnerContainer>
              <CircularProgress />
            </SpinnerContainer>
          ) : (
            <>
              <TabContent activeTab={tab}>
                {keys.map((k) => (
                  <TabPane key={k} tabId={k}>
                    {currentRootResults && currentRootResults.results && currentRootResults.results.map((item: FilmItemProps & PersonItemProps & PlanetItemProps & SpeciesItemProps & StarshipItemProps & VehicleItemProps) => (
                      <ItemAccordion currentRoot={currentRoot} item={item} />
                    ))}
                  </TabPane>
                ))}
                <StyledButtonContainer>
                  {currentRootResults.previous &&<Button onClick={() => dispatch(fetchIndividualRoot(currentRootResults.previous))} >Previous</Button>}
                  {currentRootResults.next &&<Button onClick={() => dispatch(fetchIndividualRoot(currentRootResults.next))}>Next</Button>}
                </StyledButtonContainer>
              </TabContent>
            </>
          )}
        </div>
      )}
    </div>
  );
};

export default Home;
