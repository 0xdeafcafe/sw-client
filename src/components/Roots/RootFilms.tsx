import { FilmProps, State } from "components/types";
import React from "react";
import { useSelector } from "react-redux";
import Modal from "../Modal/Modal";
import styled from "styled-components";
import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
import FavoriteIcon from "@mui/icons-material/Favorite";
import { removeFromLocalStore, SaveDataToLocalStorage } from "utils/utils";

const StyledIconContainer = styled.div`
  margin-left: 3px;
  display: flex;
  justify-content: flex-end;
  cursor: pointer;
`;

export const RootFilms = ({
  item: {
    characters,
    episode_id,
    opening_crawl,
    planets,
    producer,
    director,
    release_date,
    species,
    starships,
    title,
    vehicles,
    url,
  },
  isItemFavourite,
  setIsItemFavourite,
}: FilmProps) => {
  const branchedRoot = useSelector(
    (state: State) =>
      (state &&
        state.roots &&
        state.roots.branchedRoot &&
        state.roots.branchedRoot) ||
      []
  );

  const characterResult = () =>
    branchedRoot.map((item: { name: string }) => {
      return (
        <div>
          <p>{`Character Name: ${item.name}`}</p>
          <hr />
        </div>
      );
    });

  const planetResult = () =>
    branchedRoot.map((item: { name: string }) => (
      <div>
        <p>{`Planet Name: ${item.name}`}</p>
        <hr />
      </div>
    ));

  const speciesResult = () =>
    branchedRoot.map((item: { name: string }) => (
      <div>
        <p>{`Species Name: ${item.name}`}</p>
        <hr />
      </div>
    ));

  const starshipResult = () =>
    branchedRoot.map((item: { name: string }) => (
      <div>
        <p>{`Starship Name: ${item.name}`}</p>
        <hr />
      </div>
    ));

  const vehicleResult = () =>
    branchedRoot.map((item: { name: string }) => (
      <div>
        <p>{`Vehicle Name: ${item.name}`}</p>
        <hr />
      </div>
    ));

  return (
    <div>
      <StyledIconContainer
        onClick={() => {
          if (isItemFavourite) {
            removeFromLocalStore(url);
            setIsItemFavourite(false);
          } else {
            SaveDataToLocalStorage(url);
            setIsItemFavourite(true);
          }
        }}
      >
        {isItemFavourite ? <FavoriteIcon /> : <FavoriteBorderIcon />}
      </StyledIconContainer>
      <div>
        <p>{`Film Name: ${title}`}</p>
        <p>{`Film episode: ${episode_id}`}</p>
        <p>{`Opening Crawl: ${opening_crawl}`}</p>
        <p>{`Producer: ${producer}`}</p>
        <p>{`Director: ${director}`}</p>
        <p>{`Release Date: ${release_date}`}</p>
      </div>

      <div>
        {characters.length !== 0 && (
          <Modal
            displayName="Characters"
            item={characters}
            result={characterResult}
          />
        )}

        {planets.length !== 0 && (
          <Modal displayName="Planets" item={planets} result={planetResult} />
        )}

        {species.length !== 0 && (
          <Modal displayName="Species" item={species} result={speciesResult} />
        )}

        {starships.length !== 0 && (
          <Modal
            displayName="Starships"
            item={starships}
            result={starshipResult}
          />
        )}

        {vehicles.length !== 0 && (
          <Modal
            displayName="Vehicles"
            item={vehicles}
            result={vehicleResult}
          />
        )}
      </div>
    </div>
  );
};
