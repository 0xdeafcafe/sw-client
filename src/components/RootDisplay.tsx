import React, {useState, useEffect} from "react";
import { RootPerson } from "./Roots/RootPerson";
import { RootPlanets } from "./Roots/RootPlanets";
import { RootSpecies } from "./Roots/RootSpecies";
import { RootStarships } from "./Roots/RootStarships";
import { RootVehicles } from "./Roots/RootVehicles";
import { RootFilms } from "./Roots/RootFilms";
import { DisplayProps } from "./types";

export const RootDisplay = ({ item, item: {url}, currentRoot }: DisplayProps) => {
  const [isItemFavourite, setIsItemFavourite] = useState(false)

  useEffect(() => {
    const localStorageItems = JSON.parse(localStorage.getItem("session") || '[]')

    if(localStorageItems.includes(url)) {
      setIsItemFavourite(true)
    }
  })

  return(
  <>
    {currentRoot === "people" && <RootPerson setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "planets" && <RootPlanets setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "films" && <RootFilms setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "species" && <RootSpecies setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "vehicles" && <RootVehicles setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite}item={item} />}
    {currentRoot === "starships" && <RootStarships setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
  </>
);
}