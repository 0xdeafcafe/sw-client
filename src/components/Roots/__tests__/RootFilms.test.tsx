import { FilmProps } from "components/types";
import React from "react";
import { RootFilms } from "../RootFilms";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import { configure, mount } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  roots: {
    isLoading: true,
    isBranchedRootLoading: false,
    payload: void 0,
    error: void 0,
    currentRoot: {},
    branchedRoot: [
      {
        name: "Jabba Desilijic Tiure",
        height: "175",
        mass: "1,358",
        hair_color: "n/a",
        skin_color: "green-tan, brown",
        eye_color: "orange",
        birth_year: "600BBY",
        gender: "hermaphrodite",
        homeworld: "https://swapi.dev/api/planets/24/",
        films: [
          "https://swapi.dev/api/films/1/",
          "https://swapi.dev/api/films/3/",
          "https://swapi.dev/api/films/4/",
        ],
        species: ["https://swapi.dev/api/species/5/"],
        vehicles: [],
        starships: [],
        created: "2014-12-10T17:11:31.638000Z",
        edited: "2014-12-20T21:17:50.338000Z",
        url: "https://swapi.dev/api/people/16/",
      },
    ],
  },
};
const store = mockStore(initialState);

const props: FilmProps = {
  item: {
    characters: ["test"],
    created: "01-01-1976",
    director: "George Lucas",
    edited: "01-01-1976",
    episode_id: 0,
    opening_crawl: "far far away...",
    planets: [],
    producer: "Gary Kurtz",
    release_date: "01-01-1976",
    species: ["test"],
    starships: ["test"],
    title: "A new hope",
    url: "",
    vehicles: ["test"],
  },
  setIsItemFavourite: () => null,
  isItemFavourite: true,
};

it("renders correctly when there are no items", () => {
  const wrapper = mount(
    <Provider store={store}>
      <RootFilms {...props} />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
