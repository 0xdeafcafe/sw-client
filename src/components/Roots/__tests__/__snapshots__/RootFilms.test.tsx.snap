// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RootFilms
    isItemFavourite={true}
    item={
      Object {
        "characters": Array [
          "test",
        ],
        "created": "01-01-1976",
        "director": "George Lucas",
        "edited": "01-01-1976",
        "episode_id": 0,
        "opening_crawl": "far far away...",
        "planets": Array [],
        "producer": "Gary Kurtz",
        "release_date": "01-01-1976",
        "species": Array [
          "test",
        ],
        "starships": Array [
          "test",
        ],
        "title": "A new hope",
        "url": "",
        "vehicles": Array [
          "test",
        ],
      }
    }
    setIsItemFavourite={[Function]}
  >
    <div>
      <styled.div
        onClick={[Function]}
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "hEXTsB",
                "rules": Array [
                  "
  margin-left: 3px;
  display: flex;
  justify-content: flex-end;
  cursor: pointer;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          onClick={[Function]}
        >
          <div
            className="sc-bdVaJa hEXTsB"
            onClick={[Function]}
          >
            <ForwardRef(FavoriteIcon)>
              <ForwardRef(SvgIcon)
                data-testid="FavoriteIcon"
              >
                <MuiSvgIconRoot
                  aria-hidden={true}
                  as="svg"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                  data-testid="FavoriteIcon"
                  focusable="false"
                  ownerState={
                    Object {
                      "children": <path
                        d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                      />,
                      "color": "inherit",
                      "component": "svg",
                      "data-testid": "FavoriteIcon",
                      "fontSize": "medium",
                      "viewBox": "0 0 24 24",
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <Noop />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FavoriteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                    />
                  </svg>
                </MuiSvgIconRoot>
              </ForwardRef(SvgIcon)>
            </ForwardRef(FavoriteIcon)>
          </div>
        </StyledComponent>
      </styled.div>
      <div>
        <p>
          Film Name: A new hope
        </p>
        <p>
          Film episode: 0
        </p>
        <p>
          Opening Crawl: far far away...
        </p>
        <p>
          Producer: Gary Kurtz
        </p>
        <p>
          Director: George Lucas
        </p>
        <p>
          Release Date: 01-01-1976
        </p>
      </div>
      <div>
        <Modal
          displayName="Characters"
          item={
            Array [
              "test",
            ]
          }
          result={[Function]}
        >
          <div>
            <Styled(Button)
              color="secondary"
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <Noop />
              <Button
                className="css-bjdol9"
                color="secondary"
                onClick={[Function]}
                tag="button"
                type="button"
              >
                <button
                  aria-label={null}
                  className="css-bjdol9 btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Characters
                </button>
              </Button>
            </Styled(Button)>
            <Styled(ForwardRef(ModalUnstyled))
              BackdropComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                    [Function],
                    "",
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                }
              }
              aria-describedby="unstyled-modal-description"
              aria-labelledby="unstyled-modal-title"
              onClose={[Function]}
              open={false}
            >
              <Noop />
              <ForwardRef(ModalUnstyled)
                BackdropComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "div",
                    "__emotion_forwardProp": [Function],
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                      [Function],
                      "",
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(div)",
                    "render": [Function],
                    "withComponent": [Function],
                  }
                }
                aria-describedby="unstyled-modal-description"
                aria-labelledby="unstyled-modal-title"
                className="css-497lzk"
                onClose={[Function]}
                open={false}
              />
            </Styled(ForwardRef(ModalUnstyled))>
          </div>
        </Modal>
        <Modal
          displayName="Species"
          item={
            Array [
              "test",
            ]
          }
          result={[Function]}
        >
          <div>
            <Styled(Button)
              color="secondary"
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <Noop />
              <Button
                className="css-bjdol9"
                color="secondary"
                onClick={[Function]}
                tag="button"
                type="button"
              >
                <button
                  aria-label={null}
                  className="css-bjdol9 btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Species
                </button>
              </Button>
            </Styled(Button)>
            <Styled(ForwardRef(ModalUnstyled))
              BackdropComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                    [Function],
                    "",
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                }
              }
              aria-describedby="unstyled-modal-description"
              aria-labelledby="unstyled-modal-title"
              onClose={[Function]}
              open={false}
            >
              <Noop />
              <ForwardRef(ModalUnstyled)
                BackdropComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "div",
                    "__emotion_forwardProp": [Function],
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                      [Function],
                      "",
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(div)",
                    "render": [Function],
                    "withComponent": [Function],
                  }
                }
                aria-describedby="unstyled-modal-description"
                aria-labelledby="unstyled-modal-title"
                className="css-497lzk"
                onClose={[Function]}
                open={false}
              />
            </Styled(ForwardRef(ModalUnstyled))>
          </div>
        </Modal>
        <Modal
          displayName="Starships"
          item={
            Array [
              "test",
            ]
          }
          result={[Function]}
        >
          <div>
            <Styled(Button)
              color="secondary"
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <Noop />
              <Button
                className="css-bjdol9"
                color="secondary"
                onClick={[Function]}
                tag="button"
                type="button"
              >
                <button
                  aria-label={null}
                  className="css-bjdol9 btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Starships
                </button>
              </Button>
            </Styled(Button)>
            <Styled(ForwardRef(ModalUnstyled))
              BackdropComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                    [Function],
                    "",
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                }
              }
              aria-describedby="unstyled-modal-description"
              aria-labelledby="unstyled-modal-title"
              onClose={[Function]}
              open={false}
            >
              <Noop />
              <ForwardRef(ModalUnstyled)
                BackdropComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "div",
                    "__emotion_forwardProp": [Function],
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                      [Function],
                      "",
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(div)",
                    "render": [Function],
                    "withComponent": [Function],
                  }
                }
                aria-describedby="unstyled-modal-description"
                aria-labelledby="unstyled-modal-title"
                className="css-497lzk"
                onClose={[Function]}
                open={false}
              />
            </Styled(ForwardRef(ModalUnstyled))>
          </div>
        </Modal>
        <Modal
          displayName="Vehicles"
          item={
            Array [
              "test",
            ]
          }
          result={[Function]}
        >
          <div>
            <Styled(Button)
              color="secondary"
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <Noop />
              <Button
                className="css-bjdol9"
                color="secondary"
                onClick={[Function]}
                tag="button"
                type="button"
              >
                <button
                  aria-label={null}
                  className="css-bjdol9 btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Vehicles
                </button>
              </Button>
            </Styled(Button)>
            <Styled(ForwardRef(ModalUnstyled))
              BackdropComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                    [Function],
                    "",
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                }
              }
              aria-describedby="unstyled-modal-description"
              aria-labelledby="unstyled-modal-title"
              onClose={[Function]}
              open={false}
            >
              <Noop />
              <ForwardRef(ModalUnstyled)
                BackdropComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "div",
                    "__emotion_forwardProp": [Function],
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                      [Function],
                      "",
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(div)",
                    "render": [Function],
                    "withComponent": [Function],
                  }
                }
                aria-describedby="unstyled-modal-description"
                aria-labelledby="unstyled-modal-title"
                className="css-497lzk"
                onClose={[Function]}
                open={false}
              />
            </Styled(ForwardRef(ModalUnstyled))>
          </div>
        </Modal>
      </div>
    </div>
  </RootFilms>
</Provider>
`;
